Can't build from source with gcc-10 (Version: 1.12.0)

<!-- Please search existing issues to avoid creating duplicates.-->

## Issue Description
Can't build with latest release

## Steps to Reproduce
```
git clone https://github.com/xrplf/rippled
git checkout 89780c8e4fd4d140fcb912cf2d0c01c1b260539e
conan export external/snappy snappy/1.1.10@
conan export external/soci soci/4.0.3@
mkdir .build && cd .build
conan install .. --output-folder . --build missing --settings build_type=Release
cmake -DCMAKE_TOOLCHAIN_FILE:FILEPATH=build/generators/conan_toolchain.cmake -DCMAKE_BUILD_TYPE=Release ..
cmake --build . -- --quiet
```

## Expected Result
a built rippled binary

## Actual Result
```
#13 441.1 [ 10%] Building CXX object CMakeFiles/xrpl_core.dir/src/ripple/protocol/impl/SField.cpp.o
#13 442.0 [ 10%] Building CXX object CMakeFiles/xrpl_core.dir/src/ripple/protocol/impl/SOTemplate.cpp.o
#13 443.0 [ 10%] Building CXX object CMakeFiles/xrpl_core.dir/src/ripple/protocol/impl/STAccount.cpp.o
#13 445.7 [ 10%] Building CXX object CMakeFiles/xrpl_core.dir/src/ripple/protocol/impl/STAmount.cpp.o
#13 449.2 /build/ripple/binary/src/ripple/protocol/impl/STAmount.cpp: In lambda function:
#13 449.2 /build/ripple/binary/src/ripple/protocol/impl/STAmount.cpp:1577:15: error: expected nested-name-specifier before 'enum'
#13 449.2  1577 |         using enum Number::rounding_mode;
#13 449.2       |               ^~~~
#13 449.2 /build/ripple/binary/src/ripple/protocol/impl/STAmount.cpp:1579:40: error: 'upward' was not declared in this scope
#13 449.2  1579 |             roundUp ^ resultNegative ? upward : downward);
#13 449.2       |                                        ^~~~~~
#13 449.2 /build/ripple/binary/src/ripple/protocol/impl/STAmount.cpp:1579:49: error: 'downward' was not declared in this scope
#13 449.2  1579 |             roundUp ^ resultNegative ? upward : downward);
#13 449.2       |                                                 ^~~~~~~~
#13 449.3 /build/ripple/binary/src/ripple/protocol/impl/STAmount.cpp: In instantiation of 'ripple::STAmount ripple::divRoundImpl(const ripple::STAmount&, const ripple::STAmount&, const ripple::Issue&, bool) [with MightSaveRound = ripple::{anonymous}::DontAffectNumberRoundMode]':
#13 449.3 /build/ripple/binary/src/ripple/protocol/impl/STAmount.cpp:1609:76:   required from here
#13 449.3 /build/ripple/binary/src/ripple/protocol/impl/STAmount.cpp:1578:30: warning: unused variable 'savedRound' [-Wunused-variable]
#13 449.3  1578 |         MightSaveRound const savedRound(
#13 449.3       |                              ^~~~~~~~~~
#13 450.1 make[2]: *** [CMakeFiles/xrpl_core.dir/build.make:863: CMakeFiles/xrpl_core.dir/src/ripple/protocol/impl/STAmount.cpp.o] Error 1
#13 450.1 make[1]: *** [CMakeFiles/Makefile2:120: CMakeFiles/xrpl_core.dir/all] Error 2
#13 450.1 make: *** [Makefile:130: all] Error 2
```

## Environment
Ubuntu 20.04
CONAN_RELEASE_VERSION=1.59.0

--- a/BUILD.md
+++ b/BUILD.md
@@ -41,9 +41,9 @@ The [minimum compiler versions][2] required are:
 
 | Compiler    | Version |
 |-------------|---------|
-| GCC         | 10      |
+| GCC         | 11      |
 | Clang       | 13      |
-| Apple Clang | 13.1.6  |
+| Apple Clang | 14.0.3  |
 | MSVC        | 19.23   |
 
 We don't recommend Windows for `rippled` production at this time. As of

--- a/BUILD.md
+++ b/BUILD.md
@@ -43,7 +43,7 @@ The [minimum compiler versions][2] required are:
 |-------------|---------|
 | GCC         | 11      |
 | Clang       | 13      |
-| Apple Clang | 14.0.3  |
+| Apple Clang | 13.1.6  |
 | MSVC        | 19.23   |
 
 We don't recommend Windows for `rippled` production at this time. As of

